### ISSUE TITLE
Request - Global variables that load in-line with documentation text

### ISSUE THREAD

**Issue by dsaundersWizmo** [BookStackApp/BookStack#821]
### For Feature Requests

Desired Feature:

1. New table added to bookstack which contains variable names and values.
2. New page added somewhere to manage the table of variables, though dev ops teams can just use the table too.
3. On page load, variable placeholders will be replaced with their match.
4. Some convention (such as being wrapped in @@) to determine variable names to load.

**Example:**
_Page Content when editing:_
"Be sure to verify that the current version of Software X is @software_x_current_version@ before adding the server to production."

_Database Content:_
id |                name                    | value
1  | software_x_current_version | 11.0.8

_Displayed content:_
"Be sure to verify that the current version of Software X is 11.0.8 before adding the server to production."

This would work inline with text rather than having to reference an entire bookmarked section.  This would also be great because teams can automate updates to specified tables to have current documentation of items without having to manually update pieces.

**Comment by mendiromania** [comment]
👍

**Comment by SonGokussj4** [comment]
This would be so cool...
Right now I'm creating new documentation about the path where we have our scripts saved. 

So now it's `/mnt/current/path/scripts` but in the future it can be `/server2/final/scripts` so I would need to re-edit all pages/chapters where I use this variable.

Something like `@SCRIPTPATH@/scripts` would be awesome.

**Comment by homer2320776** [comment]
+1 for this

**Comment by Obinoben** [comment]
+1 

It would be nice to include default variable placeholders like the revision number of the page because we need to display it on our procedures. Some like @current_page_revision@ ;)

**Comment by ssddanbrown** [comment]
Just to confirm, if not aware, there is a page include system which often be used for these scenarios:
https://www.bookstackapp.com/docs/user/reusing-page-content/

**Comment by SonGokussj4** [comment]
Yes, I know about that and tried to use that. It could help by creating some "variables" page within "global" book or something like that and then on each line specify "variable", which would be then linked in different pages.
But as I tried things some weeks ago, I believe there was a problem with that - it inserts the block into a new book. Not in the middle of text for example. 
So the foundations are there. But some specific tab in settings, where you specify book-wise or site-wise variables that can be inserted anywhere (and they would be respecting formatting), would be great :-)

**Comment by bhsmither** [comment]
My experiments (that had a different problem) showed that
```<p id="bkmrk-this-year">2022</p>```
which allowed for inline placement
```Copyright {{@99#bkmrk-this-year}}, Foo Corp.<br>All rights reserved.```
resulting in
```
Copyright 2022, Foo Corp.
All rights reserved.
```
It was a single experiment. I first tried `<span>` tags, expecting the result should be inline because a `<p>` tag is a block-level element. My conclusion is that it doesn't matter (from one experiment) what tag is used.

**Comment by ssddanbrown** [comment]
Yeah, I built the include system with in-line content in mind.

In most cases BookStack will use the child content of the referenced tag (Inner HTML) unless the tag is one of a few (Tables, Lists, maybe others) where that would cause issues, in which case the entire tag content (Outer HTML) will be used.

**Comment by terion-name** [comment]
upping this. very useful feature

**Comment by ShowMeIT** [comment]
+1 would be a useful feature.

**Comment by leichti** [comment]
+1 from my side. Would be extremely nice for my use case as well.

**Comment by ssddanbrown** [comment]
BookStack is now [managed on Codeberg](https://codeberg.org/bookstack/bookstack).

As part of this change, we're closing all existing GitHub issues and pull requests.
You'll instead find this issue on Codeberg here: 
https://codeberg.org/bookstack/bookstack/issues/821
  
Any further discussion for this will now be done on Codeberg.
Therefore this GitHub thread will be locked.
